### PR TITLE
Reorganize test code for simple and clear using async / await

### PR DESCRIPTION
**Pull request by MichaelKim20**
I created this to simplify the complexity of the asynchronous function and ensure that accurate testing is done.
And I refer to the method shown in the link below.
https://www.testim.io/blog/testing-promises-using-mocha/

### CODE REVIEW

File: tests/Recovery.test.ts [hunk 1]
```diff
@@ -151,177 +151,117 @@ describe ('Test of Recovery', () =>
...
-                for (let idx = 0; idx <= 4; idx++)
-                {
-                    let uri = URI(stoa_addr)
-                        .directory("block")
-                        .addSearch("block_height", idx);
+        await client.post(url, {block: recovery_sample_data[0]});
+        await client.post(url, {block: recovery_sample_data[1]});
+        // Blocks 2 is recovered, Block 3 is saved
+        await client.post(url, {block: recovery_sample_data[3]});
+        await client.post(url, {block: recovery_sample_data[4]});
```
Comment (TrustHenry) [inline tests/Recovery.test.ts:189]: The test code order has been changed?

Comment (MichaelKim20) [reply tests/Recovery.test.ts:189]: The source code below is the original code.
The code was simplified using the function delay(), and the github may not be detected due to a large amount of variation
``` TypeScript
(async () =>
        {
            agora_node.delay = 100;

            let uri = URI(stoa_addr)
                .directory("block_externalized");

            let url = uri.toString();

            await client.post(url, {block: recovery_sample_data[0]});

            // Delay to wait for data added to the pool to be processed.
            setTimeout(() =>
            {
                client.post(url, {block: recovery_sample_data[1]});
            }, 15);

            // Blocks 2 is recovered, Block 3 is saved
            setTimeout(() =>
            {
                client.post(url, {block: recovery_sample_data[3]});
            }, 30);

            // Make sure Block 4 arrives during the saving of Block 2, and 3.
            // 100ms - 30ms < 100ms(agora_node.delay)
            setTimeout(() =>
            {
                client.post(url, {block: recovery_sample_data[4]});
            }, 100);

            // Block 3 is ignored.
            // If Block 3 was not ignored and attempted to write
            // to the database, an error would occur.
            setTimeout(() =>
            {
                client.post(url, {block: recovery_sample_data[3]});
            }, 130);

            setTimeout(async () =>
            {
                // Verifies that all sent blocks are wrote
                for (let idx = 0; idx <= 4; idx++)
                {
                    let uri = URI(stoa_addr)
                        .directory("block")
                        .addSearch("block_height", idx);

                    let response = await client.get(uri.toString());
                    assert.strictEqual(response.status, 200);
                    assert.strictEqual(response.data.height, idx);
                }

                doneIt();

            }, 300);
        })();
```

Comment (MichaelKim20) [reply tests/Recovery.test.ts:189]: Visual Studio Code detects well.

**Review by TrustHenry — APPROVED** [review]
LGTM

**Pull request merged**